pandas/tests/indexes/multi/test_drop.py

```diff
@@ -147,3 +147,22 @@ def test_drop_with_nan_in_index(nulls_fixture):
     msg = r"labels \[Timestamp\('2001-01-01 00:00:00'\)\] not found in level"
     with pytest.raises(KeyError, match=msg):
         mi.drop(pd.Timestamp("2001"), level="date")
+
+
+def test_single_level_drop():
+    # GH 37820
+
+    mi = MultiIndex.from_tuples([(1, 2), (2, 2), (3, 2)])
+    msg = r"labels \[4\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        mi.drop(4, level=0)
+    with pytest.raises(KeyError, match=msg):
+        mi.drop([1, 4], level=0)
+    msg = r"labels \[nan\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        mi.drop([np.nan], level=0)
+
```

Comment:
Could you add a case, where a list of elements should be dropped where nan is the only missing element? This is a bit of an edge case after our changes, but should work as far as I can see

Reply:
done